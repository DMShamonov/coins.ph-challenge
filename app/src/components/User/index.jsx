// @flow

import * as React from 'react';
import classnames from 'classnames';

import Avatar from 'components/Avatar';

import type { PropsType } from './types.js.flow';

export default function User({
  name,
  avatar = '',
  className,
}: PropsType): React.Node {
  return (
    <section className={classnames('user', className)}>
      <Avatar avatar={avatar} className="user__avatar" />
      <p className="user__name">{name}</p>
    </section>
  );
}
