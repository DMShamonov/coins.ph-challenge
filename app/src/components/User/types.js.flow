// @flow

export type PropsType = {|
  name: string,
  avatar?: string,
  className?: string,
|};
