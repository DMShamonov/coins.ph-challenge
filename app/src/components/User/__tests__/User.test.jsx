import React from 'react';
import { shallow } from 'enzyme';

import User from '../index';

describe('`User` component should render correctly', () => {
  test('with default props', () => {
    expect(shallow(<User name="__USER_NAME__" />)).toMatchSnapshot();
  });

  test('with `avatar` prop', () => {
    expect(shallow(<User name="__USER_NAME__" avatar="__AVATAR__" />)).toMatchSnapshot();
  });

  test('with `className` prop', () => {
    expect(shallow(<User name="__USER_NAME__" className="__CLASSNAME__" />)).toMatchSnapshot();
  });
});
