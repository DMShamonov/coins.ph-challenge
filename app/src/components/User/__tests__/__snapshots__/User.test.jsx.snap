// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`\`User\` component should render correctly with \`avatar\` prop 1`] = `
<section
  className="user"
>
  <Avatar
    avatar="__AVATAR__"
    className="user__avatar"
  />
  <p
    className="user__name"
  >
    __USER_NAME__
  </p>
</section>
`;

exports[`\`User\` component should render correctly with \`className\` prop 1`] = `
<section
  className="user __CLASSNAME__"
>
  <Avatar
    avatar=""
    className="user__avatar"
  />
  <p
    className="user__name"
  >
    __USER_NAME__
  </p>
</section>
`;

exports[`\`User\` component should render correctly with default props 1`] = `
<section
  className="user"
>
  <Avatar
    avatar=""
    className="user__avatar"
  />
  <p
    className="user__name"
  >
    __USER_NAME__
  </p>
</section>
`;
